.github/workflows/ci.yml

```diff
@@ -246,7 +246,7 @@ jobs:
     - name: cmake
       run: |
         mkdir -p out
-        cmake -S . -B out -G Ninja -DCMAKE_INSTALL_PREFIX=out/install -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_C_FLAGS="$COMPILER_FLAGS" -DCMAKE_CXX_FLAGS="$COMPILER_FLAGS" -DCMAKE_EXE_LINKER_FLAGS="$LINKER_FLAGS"
+        cmake -S . -B out -G Ninja -DCMAKE_INSTALL_PREFIX=out/install -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_C_FLAGS="$COMPILER_FLAGS" -DCMAKE_CXX_FLAGS="$COMPILER_FLAGS" -DCMAKE_SHARED_LINKER_FLAGS="$LINKER_FLAGS" -DCMAKE_EXE_LINKER_FLAGS="$LINKER_FLAGS"
```

Comment:
Is this change actually related? the linker flags aren't actually changing here?

Reply:
Yup, I'm adding `-Wl,--no-undefined`.   This new flag causes the asan build of libbinaryen.so to fail unless `-fsantize=address` is passed (which adds the asan libs to the link).

Reply:
I guess I'm confused. You're adding `-Wl,--no-undefined` in the CMakeLists.txt but you also need something on the command line for it to work?

Reply:
It turns out that `-Wl,--no-undefined` is not compatible with `-fsanitize=address`.

But what I was trying to do where waws to add `-fsanitize=address` to the link of `libbinaryen.so`.. currently the `-fsanitize=address` gets injected from the outside here in the github CI.. but its only injected when linked executables, not when linking shared libraries.

Reply:
OK, what you have now makes sense to me.